src/osgAnimation/RigTransformHardware.cpp

```diff
@@ -333,11 +347,10 @@ bool RigTransformHardware::init(RigGeometry& rig){
         stateset->addUniform(_uniformMatrixPalette);
 
         stateset->removeAttribute(osg::StateAttribute::PROGRAM);
-        if(!stateset->getAttribute(osg::StateAttribute::PROGRAM))
-            stateset->setAttributeAndModes(program.get());
+        stateset->setAttributeAndModes(program.get());
```

Comment:
The removeAttribute could removed as the setAttributes() will be replace the previous one automatically.  The setAttributeAndModes() could be replaced with setAttribute() as there aren't any GL modes associated with osg/glProgram,